### PR TITLE
Turn templates to operators

### PR DESCRIPTION
**Pull request by elronbandel**

### CODE REVIEW

File: src/unitxt/templates.py [hunk 1]
```diff
@@ -19,56 +16,30 @@ def get_postprocessors(self) -> List[str]:
...
-
-    @abstractmethod
-    def process_outputs(self, outputs: Dict[str, object]) -> Dict[str, object]:
-        pass
-
-    @abstractmethod
-    def get_postprocessors(self) -> List[str]:
-        pass
+class Template(StreamInstanceOperator):
+    """The rule of template is to take the fields of every instance and verbalize it.
```
Comment (yoavkatz) [inline src/unitxt/templates.py:20]:
```suggestion
    """The role of template is to take the fields of every instance and verbalize it.
```

File: src/unitxt/templates.py [hunk 2]
```diff
@@ -221,11 +122,21 @@ class YesNoTemplate(Template):
     label_field: str = None
     yes_answer: str = "Yes"
     no_answer: str = "No"
+    yes_answer_references: List[str] = ["Yes", "yes", "1", "True", "true"]
```
Comment (yoavkatz) [inline src/unitxt/templates.py:125]: To be consist for now, let set  

yes_answer_references: List[str]  = ["Yes"]
no_answer_references: List[str]  = ["No"]

and discussion additions in future PRs.

File: tests/test_templates.py [hunk 3]
```diff
@@ -181,7 +196,7 @@ def test_yes_no_template_process_output(self):
             yes_answer: {label_field: ["news", "sports"], class_field: ["news"]},
         }
         for expected_processed_output, outputs in processed_output_to_outputs.items():
-            processed = template.process_outputs(outputs)
+            processed, _ = template.outputs_to_target_and_references(outputs)
```
Comment (yoavkatz) [inline tests/test_templates.py:199]: Should add checks for references.

File: src/unitxt/catalog/cards/piqa.json [hunk 4]
```diff
@@ -98,9 +98,17 @@
...
+            "field": "choices_list",
+            "to_field": "choices_list",
+            "add": [
+                ""
+            ]
+        },
         {
             "type": "join_str",
-            "separator": " ",
+            "separator": "\n",
```
Comment (yoavkatz) [inline src/unitxt/catalog/cards/piqa.json:111]: Why this change?  Does anyone use this dataset?

File: src/unitxt/catalog/instructions/models/llama.json [hunk 5]
```diff
@@ -1,4 +1,4 @@
 {
     "type": "textual_instruction",
-    "text": "<<SYS>>\nYou are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.\n\nIf a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don\u2019t know the answer to aquestion, please don\u2019t share false information.\n<</SYS>>\n\n"
+    "text": "<<SYS>>\nYou are a helpful, respectful and honest assistant. Always answer as helpfully as possible, while being safe. Your answers should not include any harmful, unethical, racist, sexist, toxic, dangerous, or illegal content. Please ensure that your responses are socially unbiased and positive in nature.\n\nIf a question does not make any sense, or is not factually coherent, explain why instead of answering something not correct. If you don't know the answer to aquestion, please don't share false information.\n<</SYS>>\n\n"
```
[the remark this answers is not in the document]
Comment (yoavkatz) [reply src/unitxt/catalog/instructions/models/llama.json:3]: I suggest leaving this change out.  Doing it in a separate PR.

**Review by yoavkatz — CHANGES_REQUESTED** [review]
Basically, I think this PR should be clean and self contained.
Do not include changes that can effect results (e.g. changes to datasets or prompts)

File: src/unitxt/templates.py (hunk 2 above)
Comment (elronbandel) [reply src/unitxt/templates.py]: changed to match exactly the original behavior

File: tests/test_templates.py (hunk 3 above)
Comment (elronbandel) [reply tests/test_templates.py:199]: I would do it in a separate PR as this one does not aim to add things just to reorganize code

File: src/unitxt/catalog/cards/piqa.json (hunk 4 above)
Comment (elronbandel) [reply src/unitxt/catalog/cards/piqa.json]: The change was deleted and will be resolved in a separate PR

**Comment by elronbandel** [comment]
This PR was reduced to changes that related only for the purpose of this PR.

File: tests/test_templates.py [hunk 6]
```diff
@@ -260,101 +316,122 @@ def _test_with_wrong_class_value(wrong_class_value):
...
+        check_operator(template, inputs, targets, tester=self)
 
     def test_span_labeling_json_template(self):
-        postprocessor1, _ = fetch_artifact("processors.load_json")
-        postprocessor2, _ = fetch_artifact(
-            "processors.dict_of_lists_to_value_key_pairs"
-        )
-
-        template = SpanLabelingJsonTemplate()
+        template = SpanLabelingJsonTemplate(input_format="{text}")
```
Comment (yoavkatz) [inline tests/test_templates.py:378]: Why was this change needed?

**Review by yoavkatz — APPROVED** [review]
Hi. I approved but I had one question about a change and also have a request.

I'd  like you to work with Matan before merging  to see that FM-eval works (e.g. run with main branch and then this branch) and see it works.
